cmd/cluster/etcd_health.go

```diff
@@ -0,0 +1,239 @@
+	}
+
+	fmt.Println("+----------------------------------------------------------------+")
+	fmt.Printf("|                CONTROLPLANE NODE STATUS                        |\n")
+	fmt.Println("+----------------------------------------------------------------+")
+
+	for _, node := range nodeList.Items {
+		for _, condition := range node.Status.Conditions {
+			if condition.Type == corev1.NodeReady {
+				fmt.Printf("%s\t%s\t%s\n", node.Name, corev1.NodeReady, condition.Status)
+			}
+		}
+	}
+	return nil
+}
+
+func EtcdPodStatus(kubeCli client.Client, flags *genericclioptions.ConfigFlags) (*corev1.PodList, error) {
+
+	pods := &corev1.PodList{}
+	if err := kubeCli.List(context.TODO(), pods, client.InNamespace(EtcdNamespceNAME), client.MatchingLabels{EtcdPodMatchLabelName: EtcdPodMatchValueName}); err != nil {
+		return nil, err
+	}
+
+	fmt.Println("+----------------------------------------------------------------+")
+	fmt.Printf("|               ETCD POD STATUS                                  |\n")
+	fmt.Println("+----------------------------------------------------------------+")
+
+	for _, pods := range pods.Items {
+		containerReadyCount := 0
+		for _, containers := range pods.Status.ContainerStatuses {
+			if containers.Ready {
+				containerReadyCount++
+			}
+		}
+		fmt.Printf("%s\t%v/%v\n", pods.Name, containerReadyCount, len(pods.Status.ContainerStatuses))
+	}
+	return pods, nil
+}
+
+func EtcdCrStatus(kubeCli client.Client) error {
+
+	etcdCR := &operatorv1.Etcd{}
+	if err := kubeCli.Get(context.TODO(), client.ObjectKey{
+		Name: "cluster",
+	}, etcdCR); err != nil {
+		return err
+	}
+
+	etcdConditionList := etcdCR.Status.Conditions
+	for _, v := range etcdConditionList {
+		if v.Type == EtcdMemberConditionType {
+			fmt.Println("+---------------------------------------------------------------+")
+			fmt.Printf("|               ETCD MEMBER HEALTH STATUS                       |\n")
+			fmt.Println("+---------------------------------------------------------------+")
+			fmt.Printf("%s\n\n", v.Message)
+		}
+	}
+	return nil
+}
+
+func Etcdctlhealth(kconfig *rest.Config, clientset *kubernetes.Clientset, etcdctlCmd string, etcdPodName string) (string, error) {
+
+	cmd := []string{
+		"sh",
+		"-c",
+		etcdctlCmd,
+	}
+	req := clientset.CoreV1().RESTClient().Post().Resource("pods").Name(etcdPodName).
+		Namespace(EtcdNamespceNAME).SubResource("exec")
+	option := &corev1.PodExecOptions{
+		Container: ConatinerName,
+		Command:   cmd,
+		Stdin:     true,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       true,
+	}
+
+	if os.Stdin == nil {
+		option.Stdin = false
+	}
+	req.VersionedParams(
+		option,
+		scheme.ParameterCodec,
+	)
+
+	exec, err := remotecommand.NewSPDYExecutor(kconfig, "POST", req.URL())
+	if err != nil {
+		return "", err
+	}
+
+	capture := &logCapture{}
+	errorCapture := &logCapture{}
+
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdin:  os.Stdin,
+		Stdout: capture,
+		Stderr: errorCapture,
+		Tty:    false,
+	})
```

Comment:
To avoid using deprecated functions, let's switch to this:
```suggestion
	err = exec.StreamWithContext(context.TODO(), remotecommand.StreamOptions{
		Stdin:  os.Stdin,
		Stdout: capture,
		Stderr: errorCapture,
		Tty:    false,
	})
```

Reply:
This has been addressed in my latest commit.